pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -339,6 +340,26 @@ func (gc *GarbageCollector) isDangling(reference metav1.OwnerReference, item *no
 		return false, nil, err
 	}
 
+	if reference.Namespace != nil {
```

Comment:
I thought we were going to make the gc skip gc.apiResource if the reference.Namespace and reference.Resource are specified, but the implementation here is using reference.Namespace and reference.Resource for additional validation.

Reply:
> I thought we were going to make the gc skip gc.apiResource if the reference.Namespace and reference.Resource are specified, but the implementation here is using reference.Namespace and reference.Resource for additional validation.

Yes.  I will update, but perhaps in a follow-on pull.  I'll probably also leave these checks in place for cases where one or the other, but not both are specified and we want the cross check.